tsk/fs/hfs.cpp

```diff
@@ -367,7 +366,8 @@ hfs_ext_find_extent_record_attr(HFS_INFO * hfs, uint32_t cnid,
 
     // allocate a node buffer
     nodesize = tsk_getu16(fs->endian, hfs->extents_header.nodesize);
-    if ((node = (char *) tsk_malloc(nodesize)) == NULL) {
+    std::unique_ptr<char[]> node{new char[nodesize]};
+    if (!node) {
```

Comment:
can this even fail? I thought that a failed `new` would throw an exception?

Reply:
Copilot
2 minutes ago
In modern C++ implementations, new does indeed throw a std::bad_alloc exception if it fails to allocate memory. Therefore, checking the result of new against NULL or nullptr is unnecessary. The code you are referring to, which checks if the allocation was successful by examining the result of new, is outdated and does not reflect the current behavior of the new operator in C++:

std::unique_ptr<char[]> node{new char[nodesize]};
if (!node) {
    return 1;
}
This check is redundant because new will throw an exception if the allocation fails.

Reply:
The check isn't redundant. The call to `new` is to the wrong overload of the operator. We're not expecting exceptions; for the error handling code to continue working without modification, we need the nothrow version of `new` here. I've pushed a change to use that. Thanks for spotting the problem. 

It's been a while since I've worked behind a C API that doesn't have exception traps at the boundaries---which we should absolutely have---so I'd forgotten about the need to use nothrow new in cases like this.